### DISCUSSION TITLE
System crash when or after allocating memory resource for image.

### DISCUSSION THREAD

**Discussion by BingXuanWuSmc** [ipxe/ipxe#948] [Q&A]
Hi,
We encounter the exception occurs and hang on the moment, like below image.
![image_2023_04_24T09_25_13_755Z](https://user-images.githubusercontent.com/56142085/234159105-f7522f2d-6b4a-4145-b0d1-a80d8ddd0190.png)
We stress the system and it happen with lower fail rate about one time per 200~300 runs.
Do someone have the related experience on the problems?
Thanks.

**Comment by stappersg** [comment]
The screenshot shows `1.21.1+ (g1192e)`, the `1192e` is a _short git commit hash_.
```text
$ git log 1192e...1192e^1
commit 1192edf394aaebde666c6aafda1c8b782b08a845
Author: Michael Brown <mcb30@ipxe.org>
Date:   Thu Mar 11 15:54:26 2021 +0000

    [dhcp] Handle DHCPNAK by returning to discovery state
    
    Handle a DHCPNAK by returning to the discovery state to allow iPXE to
    attempt to obtain a replacement IPv4 address.
    
    Reuse the existing logic for deferring discovery when the link is
    blocked: this avoids hammering a misconfigured DHCP server with a
    non-stop stream of requests and allows the DHCP process to eventually
    time out and fail.
    
    Originally-implemented-by: Blake Rouse <blake.rouse@canonical.com>
    Signed-off-by: Michael Brown <mcb30@ipxe.org>
$
```

> Do someone have the related experience on the problems?

I can't answer that question, neither could I resist to ignore the question.

> one time per 200~300 runs

For me works **iPXE** _100 of 100_.  (I use recent version.)

> Thanks.

**Comment by NiKiZe** [comment]
What does imgstat show before the boot command?
How much memory is in the machine?

Which kernel is it?

One suggestion to try is to ensure you have imgfree before loading the initrd and kernel, also make sure that you provide the initrd parameter on the kernel command line in older versions of iPXE or Linux kernels (it is required for efi boot)
